beacon-chain/blockchain/receive_block.go

```diff
@@ -131,6 +132,168 @@ func (c *ChainService) ReceiveBlockDeprecated(ctx context.Context, block *ethpb.
 	return beaconState, nil
 }
 
+// ReceiveBlock is a function that defines the operations that are preformed on
+// blocks that is received from rpc service. The operations consists of:
+//   1. Gossip block to other peers
+//   2. Validate block, apply state transition and update check points
+//   3. Apply fork choice to the processed block
+//   4. Save latest head info
+func (c *ChainService) ReceiveBlock(ctx context.Context, block *ethpb.BeaconBlock) error {
+	ctx, span := trace.StartSpan(ctx, "beacon-chain.blockchain.ReceiveBlock")
+	defer span.End()
+
+	// Broadcast the new block to the network.
+	if err := c.p2p.Broadcast(ctx, block); err != nil {
+		return errors.Wrap(err, "could not broadcast block")
+	}
+
+	// Apply state transition on the new block.
+	if err := c.forkChoiceStore.OnBlock(ctx, block); err != nil {
```

Comment:
Can you call `c.ReceiveBlockNoPubSub(ctx, block)` here? this should all be the same right?